addons/account/models/partner.py

```diff
@@ -434,6 +434,17 @@ def _compute_bank_count(self):
         for partner in self:
             partner.bank_account_count = mapped_data.get(partner.id, 0)
 
+    @api.constrains('company_id')
+    def _check_company_and_amls(self):
+        amls = self.env['account.move.line'].sudo().search([
+            ('partner_id', 'in', self.ids),
+            ('reconciled', '=', False),
+            ('account_id.deprecated', '=', False),
+            ('move_id.state', '=', 'posted'),
+        ])
```

Comment:
Do a search with `limit=1` and add a constrain on the company in the domain instead. No need to fetch all these amls in the cache.

Reply:
Indeed, I added the `limit`
However, not sure about the company constraint: suppose we are updating one partner from company A, another from company B, both to company C